kotachang/soundinator
# Switched the project to C++

### Description
Switched the project language to C++. Functionality should remain the same to the user (phone or python script)

### Changes in the codebase
CMakeLists.txt modified to accomodate changes to the files
soundinator_main.c -> soundinator_main.cpp
nimble/ble functions inside soundinator_main.c -> restructured as the Nimbler class in nimbler.cpp and nimbler.hpp
code sequence (flow of function calls) remains the same

### Testing instructions
1. Build and flash from the original branch
2. Run the python_client.py script as the following: `python python_client.py --name Soundinator face` (this specifies the device name as Soundinator, and face is the hex representation of the characteristic uuid we want to get notified from)
3. Take note of the output from the script
4. Build and flash from this branch (with the C++ code)
5. Repeat steps 2-3 and check that the output is identical to the main branch
